honeycombio/honeycomb-opentelemetry-go
# maint: add smoke tests for traces

## Which problem is this PR solving?

- updates #15 

## Short description of the changes

- add smoke test infra similar to other distros (basic smoke tests as bats files, Makefile, docker-compose, Dockerfile for one of the example apps)
- tests the auto-instrumentation
- test http and grpc protocols

## How to verify that this has the expected result
  
run `make smoke` locally and see passing tests along with local json files of output in `smoke-tests/collector/data-results/data-app-sdk-grpc.json` and `smoke-tests/collector/data-results/data-app-sdk-http.json`

then `make unsmoke` to tear everything down.

note: As a follow-up, we should review how endpoints are set because it should not require removal of `http://`. Also we should make sure it is as easy as possible (and clearly documented) to set the insecure option for similar use cases of sending to an endpoint within the network. Finally, this PR does not currently add smoke tests to circle, so that should be added as well.
